src/Glpi/Form/Destination/CommonITILField/EntityFieldStrategy.php

```diff
@@ -127,4 +136,71 @@ public function getEntityIDForLastValidAnswer(
 
         return (int) $value['items_id'];
     }
+
+    public function getEntityIdFromRequester(
+        array $input,
+    ): ?int {
+        // Load requesters from input
+        $requesters = $input['_users_id_requester'] ?? null;
+        if (!$requesters) {
+            return null;
+        }
+
+        // Get ID of the first requester
+        $requester_id = current($requesters);
+        $requester = User::getById($requester_id);
+        if (!$requester) {
+            return null;
+        }
+
+        // Get profiles of the requester
+        $profiles = (new Profile_User())->find([
+            'users_id' => $requester->getID(),
+        ]);
+
+        // User has no entity
+        if (count($profiles) == 0) {
+            return null;
+        }
+        $profile = current($profiles);
+
+        // If only one profile, use its entity
+        if (count($profiles) == 1) {
+            return $profile['entities_id'];
+        }
+
+        // Look for default profile
+        $default_profiles = (new Profile_User())->find([
+            'users_id' => $requester->getID(),
+            'profiles_id' => new QuerySubQuery([
+                'SELECT' => 'id',
+                'FROM'   => Profile::getTable(),
+                'WHERE'  => [
+                    'interface'  => 'helpdesk',
+                    'is_default' => 1,
+                ],
+            ]),
+        ]);
+        if (count($default_profiles) == 1) {
+            $default_profile = current($default_profiles);
+            return $default_profile['entities_id'];
+        }
+
+        // If one or more helpdesk profiles, use entity of the first profile
+        $helpdesk_profiles = (new Profile_User())->find([
+            'users_id' => $requester->getID(),
+            'profiles_id' => new QuerySubQuery([
+                'SELECT' => 'id',
+                'FROM'   => Profile::getTable(),
+                'WHERE'  => ['interface' => 'helpdesk'],
+            ]),
+        ]);
+        if (count($helpdesk_profiles) > 0) {
+            $helpdesk_profile = current($helpdesk_profiles);
+            return $helpdesk_profile['entities_id'];
+        }
+
+        // Fallback to first profile
+        return $profile['entities_id'];
+    }
```

Comment:
Is it possible to have multiple requesters? In this case, a logic similar to what is actually done in the `Ticket::getEntitiesForRequesters()` method should be used here.

Reply:
It is possible but I've made sure to only use data for the first requester.

IMO, it does not make sense to use this setting if you plan on having multiple requester as it is impossible to guess which requester entity to use.

With that in mind, I'm using the first requester to avoid unexpected errors if someone do end up doing some illogical configurations but I don't think we need to think very hard about it.

Reply:
I remember a discussion some years ago with @orthagh  about the fact that it is not possible to force the ticket creation entity for a user.
The solution I proposed was to add a `Default entity used when creating items` field on the `User`, which could have been used for forms (`formcreator` at the time), CalDAV synchronisation, etc.

I think this solution could be revived, in a later PR, to get a better control about the behaviour of the `Entity of the requester` option.